include/sparrow/buffer.hpp

```diff
@@ -226,11 +289,29 @@ namespace sparrow
         if (n != size())
         {
             buffer<T> tmp(n);
-            std::copy(data(), data() + size(), tmp.data());
+            size_type copy_size = std::min(size(), n);
+            std::copy(data(), data() + copy_size, tmp.data());
             swap(tmp);
         }
     }
 
+    template <class T>
+    void buffer<T>::resize(size_type n, value_type value)
+    {
+        size_type old_size = size();
+        resize(n);
+        if (old_size < n)
+        {
+            std::fill(data() + old_size, data() + n, value);
```

Comment:
Alternatively, one can use:

```suggestion
            std::fill_n(data() + old_size, n, value);
```